examples/chroma-news-of-the-day/news-of-the-day.ipynb

```diff
@@ -0,0 +1,270 @@
+   "id": "d994c585-af48-416e-99fa-6bf9f736c4f7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from unstructured.partition.html import partition_html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "f3ae24d4-2926-4ea3-ad12-52e372918039",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cnn_lite_url = \"https://lite.cnn.com/\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "deca76d9-8f38-4dcc-a34a-466aaba3ab45",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "elements = partition_html(url=cnn_lite_url)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "cb258d76-459e-4249-88e3-5bd340e06a32",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "links = []\n",
+    "for element in elements:\n",
+    "    if element.metadata.links is not None:\n",
+    "        relative_link = element.metadata.links[0][\"url\"][1:]\n",
+    "        if relative_link.startswith(\"2023\"):\n",
+    "            links.append(f\"{cnn_lite_url}{relative_link}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "c4189214-0cf4-4ccf-b996-bb89c1d30233",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['https://lite.cnn.com/2023/08/07/entertainment/william-friedkin-death/index.html',\n",
+       " 'https://lite.cnn.com/2023/08/07/us/alabama-boat-dock-fight-warrants/index.html',\n",
+       " 'https://lite.cnn.com/2023/08/07/us/extreme-heat-death-toll-underestimate-climate/index.html',\n",
+       " 'https://lite.cnn.com/2023/08/06/us/charles-gregory-missing-st-augustine/index.html',\n",
+       " 'https://lite.cnn.com/2023/08/07/opinions/womens-world-cup-morocco-nigeria-south-africa-jamaica-aziz/index.html']"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "links[:5]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0e340a96-e66b-431a-8416-4a9b4d2bdadf",
+   "metadata": {},
+   "source": [
+    "## Ingest individual articles with `UnstructuredURLLoader`\n",
+    "\n",
+    "Now that we have the links, we can preprocess individual news articles with `UnstructuredURLLoader`. `UnstructuredURLLoader` fetches content from the web and then uses the `unstructured` `partition` function to extract content and metadata. In this example we preprocess HTML files, but it works with other response types such as `application/pdf` as well. After calling `.load()`, the result is a list of `langchain` `Document` objects."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "cd629187-dcd7-4411-8e61-dea34bae2963",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.document_loaders import UnstructuredURLLoader\n",
+    "\n",
+    "loaders = UnstructuredURLLoader(\n",
+    "    urls=links, mode=\"elements\", show_progress_bar=True,\n",
```

Comment:
@MthwRobinson this `mode="element"` seems to create error when storing the embeddings into ChromeDB.

![Screenshot 2023-08-07 at 2 06 30 PM](https://github.com/Unstructured-IO/unstructured/assets/138828701/5df96492-89ac-4f83-b8bb-b8dea65dcd8e)

Reply:
Yeah we'll need to PR a fix for this into `langchain`. Changed to `single` mode for now so it runs through.

Reply:
From the stacktrace, this is actually a `chromadb` limitation and not a `langchain` or `unstructured` bug. If you set `mode="elements"` and use FAISS instead of Chroma, everything works. Still probably good to have an option to filter out list metadata elements so things don't break with Chroma.

```python
from langchain.vectorstores.faiss import FAISS
vectorstore = FAISS.from_documents(docs, embeddings)
```

Here's the full stacktrace for reference:

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[10], line 2
      1 embeddings = OpenAIEmbeddings()
----> 2 vectorstore = Chroma.from_documents(docs, embeddings)

File ~/repos/langchain/libs/langchain/langchain/vectorstores/chroma.py:603, in Chroma.from_documents(cls, documents, embedding, ids, collection_name, persist_directory, client_settings, client, collection_metadata, **kwargs)
    601 texts = [doc.page_content for doc in documents]
    602 metadatas = [doc.metadata for doc in documents]
--> 603 return cls.from_texts(
    604     texts=texts,
    605     embedding=embedding,
    606     metadatas=metadatas,
    607     ids=ids,
    608     collection_name=collection_name,
    609     persist_directory=persist_directory,
    610     client_settings=client_settings,
    611     client=client,
    612     collection_metadata=collection_metadata,
    613     **kwargs,
    614 )

File ~/repos/langchain/libs/langchain/langchain/vectorstores/chroma.py:567, in Chroma.from_texts(cls, texts, embedding, metadatas, ids, collection_name, persist_directory, client_settings, client, collection_metadata, **kwargs)
    539 """Create a Chroma vectorstore from a raw documents.
    540 
    541 If a persist_directory is specified, the collection will be persisted there.
   (...)
    556     Chroma: Chroma vectorstore.
    557 """
    558 chroma_collection = cls(
    559     collection_name=collection_name,
    560     embedding_function=embedding,
   (...)
    565     **kwargs,
    566 )
--> 567 chroma_collection.add_texts(texts=texts, metadatas=metadatas, ids=ids)
    568 return chroma_collection

File ~/repos/langchain/libs/langchain/langchain/vectorstores/chroma.py:208, in Chroma.add_texts(self, texts, metadatas, ids, **kwargs)
    204     embeddings_with_metadatas = (
    205         [embeddings[idx] for idx in non_empty_ids] if embeddings else None
    206     )
    207     ids_with_metadata = [ids[idx] for idx in non_empty_ids]
--> 208     self._collection.upsert(
    209         metadatas=metadatas,
    210         embeddings=embeddings_with_metadatas,
    211         documents=texts_with_metadatas,
    212         ids=ids_with_metadata,
    213     )
    214 if empty_ids:
    215     texts_without_metadatas = [texts[j] for j in empty_ids]

File ~/.pyenv/versions/chromadb/lib/python3.10/site-packages/chromadb/api/models/Collection.py:294, in Collection.upsert(self, ids, embeddings, metadatas, documents)
    275 def upsert(
    276     self,
    277     ids: OneOrMany[ID],
   (...)
    280     documents: Optional[OneOrMany[Document]] = None,
    281 ) -> None:
    282     """Update the embeddings, metadatas or documents for provided ids, or create them if they don't exist.
    283 
    284     Args:
   (...)
    291         None
    292     """
--> 294     ids, embeddings, metadatas, documents = self._validate_embedding_set(
    295         ids, embeddings, metadatas, documents
    296     )
    298     self._client._upsert(
    299         collection_id=self.id,
    300         ids=ids,
   (...)
    303         documents=documents,
    304     )

File ~/.pyenv/versions/chromadb/lib/python3.10/site-packages/chromadb/api/models/Collection.py:349, in Collection._validate_embedding_set(self, ids, embeddings, metadatas, documents, require_embeddings_or_documents)
    342 ids = validate_ids(maybe_cast_one_to_many(ids))
    343 embeddings = (
    344     validate_embeddings(maybe_cast_one_to_many(embeddings))
    345     if embeddings is not None
    346     else None
    347 )
    348 metadatas = (
--> 349     validate_metadatas(maybe_cast_one_to_many(metadatas))
    350     if metadatas is not None
    351     else None
    352 )
    353 documents = maybe_cast_one_to_many(documents) if documents is not None else None
    355 # Check that one of embeddings or documents is provided

File ~/.pyenv/versions/chromadb/lib/python3.10/site-packages/chromadb/api/types.py:172, in validate_metadatas(metadatas)
    170     raise ValueError(f"Expected metadatas to be a list, got {metadatas}")
    171 for metadata in metadatas:
--> 172     validate_metadata(metadata)
    173 return metadatas

File ~/.pyenv/versions/chromadb/lib/python3.10/site-packages/chromadb/api/types.py:140, in validate_metadata(metadata)
    138     # isinstance(True, int) evaluates to True, so we need to check for bools separately
    139     if not isinstance(value, bool) and not isinstance(value, (str, int, float)):
--> 140         raise ValueError(
    141             f"Expected metadata value to be a str, int, float or bool, got {value} which is a {type(value)}"
    142         )
    143 return metadata

ValueError: Expected metadata value to be a str, int, float or bool, got [{'text': 'CNN', 'url': '/'}] which is a <class 'list'>
```